File: python/word-count/word_count.py
import re

def word_count(str):
  str = str.decode('utf-8')
  sentence = re.sub("[\W]|_", " ", str, re.UNICODE)
  d = {}
  for word in sentence.lower().split():
    if word in d:
      d[word] += 1
    else:
      d[word] = 1
  return d

File: ruby/word-count/word_count.rb
module BookKeeping
  VERSION = 1
end

class Phrase

  def initialize(phrase)
    @phrase = phrase
  end

  def word_count
    hash = {}
    @phrase.gsub!(',', ' ')
    @phrase.downcase.split(' ').each do |word|
      while !word[0].match(/^[[:alpha:]]$/) && !word[0].match(/[0-9]/)
        word[0] = ''
      end
      while !word[-1].match(/^[[:alpha:]]$/) && !word[-1].match(/[0-9]/)
        word[-1] = ''
      end
      if hash.key?(word)
        hash[word] += 1
      else
        hash[word] = 1
      end
    end
    hash
  end

end